lib/matplotlib/axes/_axes.py

```diff
@@ -2604,6 +2605,123 @@ def barh(self, y, width, height=0.8, left=None, *, align="center",
                            align=align, **kwargs)
         return patches
 
+    def bar_label(self, container, captions=[], *, fmt="%g", position="edge",
+                  padding=0, shifting=0, **kwargs):
```

Comment:
I think we need a comment in the docs that kwargs is passed through to `annotate`.